### ISSUE TITLE
Invalid TURN credentials: What function should fail?

### ISSUE THREAD

**Issue by alvestrand** [w3c/webrtc-pc#2593]
I had an issue come up with TURN credentials recently.

Sometimes, it's possible to tell that a TURN credential is invalid just from looking at them. For instance, username fields that violate the length restriction (509) in RFC 8489.

But I couldn't figure out from our spec what function should fail.
Alternatives:

- RTCPeerConnection constructor or setConfiguration, where the parameters are passed in
- createOffer/addIceCandidate, where the candidates are used
- nowhere - just act as if the relevant TURN candidate was never generated

The last one is easiest to implement. Is it the right solution?

**Comment by juberti** [comment]
For a too long TURN username, pretty sure setConfig should fail, same as if some other invalid IceServer attribute was supplied.

**Comment by nils-ohlmeier** [comment]
I agree with @juberti that especially in a case where the error can be detected it should be raised as early as possible, thus in the constructor or setConfig. Which is also how Firefox rejects too long TURN usernames already.

"Nowhere" appears to be the worst option, as that could result in nobody noticing anything wrong until it is too late, e.g. multiple urls provided, but one is invalid but goes unnoticed.

**Comment by alvestrand** [comment]
Should be handled together with all the other failures of ICE server parsing.

**Comment by alvestrand** [comment]
4.4.1.1 says that UnknownError should be returned when "something else" fails.

**Comment by alvestrand** [comment]
section 4.4.6.6 (apply configuration) step 10.5 is where that validation should happen.
